### ISSUE TITLE
podman ask for repo when image or manifest is local but --platform is specified

### ISSUE THREAD

**Issue by akostadinov** [containers/podman#17503]
### Issue Description

When `--platform` option is supplied to `podman run`, it tries to pull remote image and only after `Ctrl+C` it reads local manifest or image. This makes it a pita to test locally built images before they are pushed to the target registry.

### Steps to reproduce the issue

Steps to reproduce the issue
1. `buildah build --jobs=2 --platform=linux/amd64,linux/ppc64le --manifest multiarch-test .`
2. podman run --platform=linux/amd64 multiarch-test

### Describe the results you received

```
  ▸ registry.fedoraproject.org/multiarch-test:latest
    registry.access.redhat.com/multiarch-test:latest
    docker.io/library/multiarch-test:latest
    quay.io/multiarch-test:latest
```
Hit `Ctrl+C`.
```
Hello Reproducer!
```

### Describe the results you expected

Expected is container to just run.
```
Hello Reproducer!
```

### podman info output

```yaml
host:
  arch: amd64
  buildahVersion: 1.28.0
  cgroupControllers:
  - cpu
  - io
  - memory
  - pids
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.1.5-1.fc36.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.1.5, commit: '
  cpuUtilization:
    idlePercent: 69.15
    systemPercent: 6.71
    userPercent: 24.14
  cpus: 8
  distribution:
    distribution: fedora
    version: "36"
  eventLogger: journald
  hostname: koTak
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1001
      size: 1
    - container_id: 1
      host_id: 10000
      size: 54330
    uidmap:
    - container_id: 0
      host_id: 1001
      size: 1
    - container_id: 1
      host_id: 10000
      size: 54321
  kernel: 6.1.7-100.fc36.x86_64
  linkmode: dynamic
  logDriver: journald
  memFree: 4395266048
  memTotal: 33396047872
  networkBackend: cni
  ociRuntime:
    name: crun
    package: crun-1.7.2-3.fc36.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 1.7.2
      commit: 0356bf4aff9a133d655dc13b1d9ac9424706cac4
      rundir: /run/user/1001/crun
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +WASM:wasmedge +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/user/1001/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    seccompProfilePath: /usr/share/containers/seccomp.json
    selinuxEnabled: true
  serviceIsRemote: false
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.2.0-0.2.beta.0.fc36.x86_64
    version: |-
      slirp4netns version 1.2.0-beta.0
      commit: 477db14a24ff1a3de3a705e51ca2c4c1fe3dda64
      libslirp: 4.6.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.3
  swapFree: 13713690624
  swapTotal: 16982728704
  uptime: 440h 4m 59.00s (Approximately 18.33 days)
plugins:
  authorization: null
  log:
  - k8s-file
  - none
  - passthrough
  - journald
  network:
  - bridge
  - macvlan
  - ipvlan
  volume:
  - local
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - docker.io
  - quay.io
store:
  configFile: /home/user/.config/containers/storage.conf
  containerStore:
    number: 19
    paused: 0
    running: 3
    stopped: 16
  graphDriverName: overlay
  graphOptions: {}
  graphRoot: /var/lib/libvirt/images/containers/user
  graphRootAllocated: 94793416704
  graphRootUsed: 66624368640
  graphStatus:
    Backing Filesystem: extfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageCopyTmpDir: /var/tmp
  imageStore:
    number: 53
  runRoot: /run/user/1001/containers
  volumePath: /var/lib/libvirt/images/containers/user/volumes
version:
  APIVersion: 4.3.1
  Built: 1668180253
  BuiltTime: Fri Nov 11 17:24:13 2022
  GitCommit: ""
  GoVersion: go1.18.7
  Os: linux
  OsArch: linux/amd64
  Version: 4.3.1
```

### Podman in a container

No

### Privileged Or Rootless

Rootless

### Upstream Latest Release

No

### Additional environment details

Example Dockerfile
```
FROM alpine
ENTRYPOINT ["/bin/sh", "-c", "echo 'Hello Reproducer!'"]
```

### Additional information

Additional information like issue happens only occasionally or issue happens with a particular architecture or on a particular setting

**Comment by vrothberg** [comment]
Thanks for reaching out, @akostadinov!

This is the same issue as https://github.com/containers/podman/issues/17063 but with a different symptom.  As outlined in https://github.com/containers/podman/issues/17063, Podman will pull the image once --platform is specified.

It is not yet clear how to fix it though.  Previous attempts have caused regressions (e.g., https://github.com/containers/common/pull/1299).  A workaround would be to use `podman run --pull=never --platform ...`.

I am closing this issue here as duplicate of https://github.com/containers/podman/issues/17063.
